packages/@aws-cdk/aws-ec2/lib/port.ts

```diff
@@ -0,0 +1,201 @@
+ */
+export class Port {
+  /**
+   * A single TCP port
+   */
+  public static tcpPort(port: number): Port {
+    return new Port({
+      ipProtocol: Protocol.TCP,
+      fromPort: port,
+      toPort: port,
+      stringRepresentation: renderPort(port),
+    });
+  }
+
+  /**
+   * A TCP port range
+   */
+  public static tcpPortRange(startPort: number, endPort: number) {
+    return new Port({
+      ipProtocol: Protocol.TCP,
+      fromPort: startPort,
+      toPort: endPort,
+      stringRepresentation: `${renderPort(startPort)}-${renderPort(endPort)}`
+    });
+  }
+
+  /**
+   * Any TCP traffic
+   */
+  public static allTcp() {
+    return new Port({
+      ipProtocol: Protocol.TCP,
+      fromPort: 0,
+      toPort: 65535,
+      stringRepresentation: 'ALL PORTS',
+    });
+  }
+
+  /**
+   * A single UDP port
+   */
+  public static udpPort(port: number): Port {
+    return new Port({
+      ipProtocol: Protocol.UDP,
+      fromPort: port,
+      toPort: port,
+      stringRepresentation: `UDP ${renderPort(port)}`,
+    });
+  }
+
+  /**
+   * A UDP port range
+   */
+  public static udpPortRange(startPort: number, endPort: number) {
+    return new Port({
+      ipProtocol: Protocol.UDP,
+      fromPort: startPort,
+      toPort: endPort,
+      stringRepresentation: `UDP ${renderPort(startPort)}-${renderPort(endPort)}`
+    });
+  }
+
+  /**
+   * Any UDP traffic
+   */
+  public static allUdp() {
+    return new Port({
+      ipProtocol: Protocol.UDP,
+      fromPort: 0,
+      toPort: 65535,
+      stringRepresentation: 'UDP ALL PORTS',
+    });
+  }
+
+  /**
+   * A specific combination of ICMP type and code
+   *
+   * @see https://www.iana.org/assignments/icmp-parameters/icmp-parameters.xhtml
+   */
+  public static icmpTypeAndCode(type: number, code: number) {
+    return new Port({
+      ipProtocol: Protocol.ICMP,
+      fromPort: type,
+      toPort: code,
+      stringRepresentation: `ICMP Type ${type} Code ${code}`
+    });
+  }
+
+  /**
+   * All codes for a single ICMP type
+   */
+  public static icmpType(type: number): Port {
+    return new Port({
+      ipProtocol: Protocol.ICMP,
+      fromPort: type,
+      toPort: -1,
+      stringRepresentation: `ICMP Type ${type}`,
+    });
+  }
+
+  /**
+   * ICMP ping (echo) traffic
+   */
+  public static icmpPing() {
+    return Port.icmpType(8);
+  }
+
+  /**
+   * All ICMP traffic
+   */
+  public static allIcmp() {
+    return new Port({
+      ipProtocol: Protocol.ICMP,
+      fromPort: -1,
+      toPort: -1,
+      stringRepresentation: 'ALL ICMP',
+    });
+  }
+
+  /**
+   * All traffic
+   */
+  public static allTraffic() {
+    return new Port({
+      ipProtocol: Protocol.ALL,
+      stringRepresentation: 'ALL TRAFFIC',
+    });
+  }
+
+  /**
+   * Whether the rule containing this port range can be inlined into a securitygroup or not.
+   */
+  public readonly canInlineRule: boolean;
+
+  constructor(private readonly props: PortProps) {
```

Comment:
private/protected? Even just to signal that it shouldn't be instantiated...

Reply:
But it can be if you need something not covered by any of the factory functions.